CRM/Utils/System/Drupal.php

```diff
@@ -163,7 +163,10 @@ public static function checkUserNameEmailExists(&$params, &$errors, $emailName =
    */
   public function getLoginURL($destination = '') {
     $query = $destination ? array('destination' => $destination) : array();
-    return url('user', array('query' => $query), TRUE);
+   	$config = CRM_Core_Config::singleton();
```

Comment:
Actually - sorry to come late to the party but - shouldn't we be passing the 'absolute' param to the drupal function?

https://api.drupal.org/api/drupal/includes!common.inc/function/url/7.x